lib/galaxy/model/__init__.py

```diff
@@ -1157,7 +1177,7 @@ def is_hda(d):
     return isinstance( d, HistoryDatasetAssociation )
 
 
-class History( object, Dictifiable, UsesAnnotations, HasName ):
+class History( HasTags, Dictifiable, UsesAnnotations, HasName ):
```

Comment:
Sure. In the meantime 37cb522 is the simple solution for the bug this addresses.